pkg/kernel/Dockerfile

```diff
@@ -25,25 +40,24 @@ COPY / /
 
 # Download and verify kernel
 # PGP keys: 589DA6B1 (greg@kroah.com) & 6092693E (autosigner@kernel.org) & 00411886 (torvalds@linux-foundation.org)
-RUN KERNEL_VERSION="$(eval echo \$KERNEL_VERSION_"$(uname -m)")" && KERNEL_MAJOR="$(echo "$KERNEL_VERSION" | cut -f1 -d.)" && \
-    KERNEL_SOURCE=${KERNEL_SOURCE:-https://www.kernel.org/pub/linux/kernel/v${KERNEL_MAJOR}.x/linux-${KERNEL_VERSION}.tar.xz} && \
-    KERNEL_SHA256_SUMS=${KERNEL_SHA256_SUMS:-https://www.kernel.org/pub/linux/kernel/v${KERNEL_MAJOR}.x/sha256sums.asc} && \
-    KERNEL_PGP2_SIGN=${KERNEL_PGP2_SIGN:-https://www.kernel.org/pub/linux/kernel/v${KERNEL_MAJOR}.x/linux-${KERNEL_VERSION}.tar.sign} && \
-    curl -fsSLO ${KERNEL_SHA256_SUMS} && \
-    gpg2 -q --import keys.asc && \
+ADD ${KERNEL_SHA256_SUMS} kernel-sums.asc
+ADD ${KERNEL_PGP2_SIGN} linux-${KERNEL_VERSION}.tar.sign
+ADD ${KERNEL_SOURCE} /tmp/linux-${KERNEL_VERSION}.tar.xz
+
```

Comment:
Can we move `COPY / /` line here to not re-run downloading in case of config change? Or buildkit will do things in background and will not run actual download here on retry?

Reply:
Yeah, that is a good call. It should be optimized for, but why depend on it when we can control it?